packages/types/lib/IStatistic.ts

```diff
@@ -0,0 +1,101 @@
+// eslint-disable-next-line
+import type { EventEmitter } from 'events';
+import type { ILink } from '@comunica/bus-rdf-resolve-hypermedia-links';
+import type { ActionContextKey } from '@comunica/core';
+import type { IActionContext } from './IActionContext';
+import type { IQuerySource } from './IQuerySource';
+
+export interface IStatistic {
+  /**
+   * Query string of tracked execution
+   */
+  query: string, 
+
+  statisticEvents: EventEmitter;
+
+  getEmitter: () => EventEmitter;
+}
+
+export interface IAggregateStatistic extends IStatistic {
```

Comment:
I wonder if we really need a dedicated interface for this?
The stats that want to do aggregation can just do it? There's nowhere where an `IAggregateStatistic` is explicitly required I guess?

Reply:
No, I guess not, if every type of statistic tracking will have a dedicated interface. It might be needed to define a `getTrackedStatistic(): any` function in the IStatistic interface then, as you would want all statistic trackers to expose their information to other parts of Comunica.

Reply:
On second thought it's not needed, as the emitters already expose all data.